### DISCUSSION TITLE
Combining Different Connectors

### DISCUSSION THREAD

**Discussion by spinndigo** [algolia/instantsearch#5471] [Q&A] [answered]
Hello, I'm fairly new to using Algolia (version 6.38.1) and I have a question surrounding connectors. I currently have a react component which is wrapped by a connector ( in this case , the `connectCurrentRefinements` HOC). This component renders one of two buttons which acts like a toggle. The button rendered either clears all refinements/query , or alters the query so all results are shown ('See All') . 

Now I'd like to add the functionality of another connector to the component, namely `connectSortBy` . What I'd like to do is in addition to the functionality described above, switch the index to a replica index. For example, let's say I'm currently on index A . When I click 'See All' using the component described above, I want to first change to index B, then update the query. Similarly, when clicking 'Clear All' , I want to switch back to index A.

This is particular and in general i'd like to know the following:

1. What is the preferred method of combining the functionality of multiple connectors? It seems to me that nesting HOC's could potentially produce multiple searches/queries. In addition, the props provided by the connectors have identical names.

2. Is there some general use connector that can provide the wrapped component with functionality from any/all connectors?

thank you :)

**Comment by Haroenv** [comment, marked as answer]
We have fixed this when switching to React InstantSearch Hooks, you can import and use multiple hooks inside one component.

For React InstantSearch (core) you indeed need to nest the connectors and map the provided props to something non-overlapping.

**Reply by spinndigo** [reply]
Thanks so much for the fast reply. I will look into nesting the connectors ( and potentially upgrading to hooks )
